tests/run_tests.py

```diff
@@ -205,14 +206,23 @@ def run_test_throw_on_fail(config: TestConfig, file_path: Path) -> None:
 
 def run_test_print_result(test_path: Path) -> TestStatus:
     assert test_path.exists()
-    config = parse_file(test_path)
 
-    test_name = str(test_path.relative_to(PARENT_DIR))
+    # Try and shorten the path if possible.
+    test_name = (
+        str(test_path.relative_to(PARENT_DIR))
+        if test_path.is_relative_to(PARENT_DIR)
+        else str(test_path)
+    )
+
+    config = parse_file(test_path)
+    if config is None:
+        print(f"SKIPPED '{test_name}' (not a test)")
```

Comment:
This might get interleaved. Better take the `io_lock`

Reply:
Woops, good spot